JavaScript/JavaScriptSDK/Util.ts

```diff
@@ -227,19 +227,18 @@ module Microsoft.ApplicationInsights {
         /**
          * generate GUID
          */
-        public static newGuid() {
-            var hexValues = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "A", "B", "C", "D", "E", "F"];
-
-            // c.f. rfc4122 (UUID version 4 = xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx)
-            var oct = "", tmp;
-            for (var a = 0; a < 4; a++) {
-                tmp = (4294967296 * Math.random()) | 0;
-                oct += hexValues[tmp & 0xF] + hexValues[tmp >> 4 & 0xF] + hexValues[tmp >> 8 & 0xF] + hexValues[tmp >> 12 & 0xF] + hexValues[tmp >> 16 & 0xF] + hexValues[tmp >> 20 & 0xF] + hexValues[tmp >> 24 & 0xF] + hexValues[tmp >> 28 & 0xF];
-            }
+        public static newId(): string {
+            var base64chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
```

Comment:
Does server side SDK (including Java and Node) treat is as a string? If yes then you should be fine.